tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7860,6 +7860,8 @@ def err_hlsl_missing_dispatchgrid_semantic : Error<
    "Broadcasting node shader '%0' with NodeMaxDispatchGrid attribute must declare an input record containing a field with SV_DispatchGrid semantic">;
 def err_hlsl_dispatchgrid_semantic_already_specified : Error<
    "a field with SV_DispatchGrid has already been specified">;
+def err_hlsl_incompatible_dispatchgrid_semantic_type : Error<
+   "SV_DispatchGrid should be 32/16 bit uint scalar or vector/array up to 3 elements">;
```

Comment:
Should we make the diagnostic print the type found? That will make it easier to understand if this were to occur under template instantiation where the type may not appear in the code.

```suggestion
   "SV_DispatchGrid can only be applied to a declaration of 32 or 16-bit integer scalar or 3-element vector or array, not %0">;
```